README.md
# 1Sheeld Android App [![Build Status](https://travis-ci.org/Integreight/1Sheeld-Android-App.svg?branch=master)](https://travis-ci.org/Integreight/1Sheeld-Android-App) #

## Overview ##

1Sheeld app is used along with our hardware shield to either control Arduino, read your smartphone's sensors in your Arduino sketch, post on social media, or even control your Android device. It enables you to scan, connect, and interact with nearby 1Sheelds using our custom protocol

## Installation ##

You can download the latest version of our app from [Google Play Store](https://play.google.com/store/apps/details?id=com.integreight.onesheeld&hl=en), or pick your preferred version directly from our repo's [releases page](https://github.com/Integreight/1Sheeld-Android-App/releases).

## Screenshots ##

![App Screenshots](http://i.imgur.com/9KqOKKu.png)

## Building ##

The repo is a generic Gradle project, it was built and tested using the latest stable version of Android Studio.

To build the project and generate the release and debug apk(s), run this command on the root of the repo:

```
.\gradlew assemble
```

## Access Tokens ##

In case you want to use any of the web services we use, you should update its access tokens in [meta_data.json](https://github.com/Integreight/1Sheeld-Android-App/blob/master/oneSheeld/src/main/assets/meta_data.json) file.

## Compatibility ##

The app should work with Android devices running version 2.3 and above.

## Contribution ##

Contributions are welcomed, please follow this pattern:
- Fork the repo.
- Open an issue with your proposed feature or bug fix.
- Commit and push code to a new branch in your forked repo.
- Submit a pull request to our *development* branch.

Don't forget to drop us an email, post on our forum, or mention us on Twitter or Facebook about what you have built using 1Sheeld, we would love to hear about it.

## Learn More ##

- [Getting started tutorial](http://www.1sheeld.com/tutorials/getting-started)
- [Arduino library documentation](http://1sheeld.com/docs/).
- Check our [1Sheeld Forums](http://www.1sheeld.com/forum) where you can post your questions and get answers.
- Explore what people have built at our [Hackster.io page](https://www.hackster.io/1sheeld/projects)
- Built custom apps using our [Android SDK](https://github.com/Integreight/1Sheeld-Android-SDK).

## Changelog ##

To see what has changed in recent versions of 1Sheeld Android App, see the [Change Log](CHANGELOG.md).

## License and Copyright ##

```
This code is free software; you can redistribute it and/or modify it
under the terms of the GNU General Public License version 3 only, as
published by the Free Software Foundation.

This code is distributed in the hope that it will be useful, but WITHOUT
ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
version 3 for more details (a copy is included in the LICENSE file that
accompanied this code).

Please contact Integreight, Inc. at info@integreight.com or post on our
support forums www.1sheeld.com/forum if you need additional information
or have any questions.
```


CHANGELOG.md
## v1.9.0(170521) (May 21, 2017)
 - Added the face detection shield.
 - Added a button to the accelerometer shield to enable linear acceleration.
 - Fixed voice recognition shield bugs.
 - Fixed some bugs in the terminal shield where the messages weren't displayed correctly.
 - Increased overall app stability and fixed other major bugs.

## v1.8.2(170216) (February 16, 2017)
 - Changed 1Sheeld buying link to Amazon in case the user is from the US.

## v1.8.1(170112) (January 12, 2017)
 - Added a link to our website to ease the process of buying 1Sheeld boards for first users.

## v1.8.0(161113) (November 13, 2016)
 - Supported Android Nougat (v7.0), and fixed a bug where it didn't connect or disconnect sometimes.
 - Handled a bug where any files opened using a external app crashes our app.
 - Made the text to speech shield use the current locale for speaking.
 - Integrated Firebase for push notifications.
 - Changed our old logo with the brand new one.
 - Increased app stability and fixed minor bugs.

## v1.7.0(160529) (May 29, 2016)
 - Supported our new board, 1Sheeld+ (The BLE version)
 - Built the app on our own 1Sheeld Android SDK.
 - Prepared the app for translation.
 - Increased app stability and fixed minor bugs.

## v1.6.0(160126) (January 26, 2016)
 - Added vibration shield.
 - Reactivated Facebook shield again.
 - Supported Android M permissions model.
 - Added a demo mode for exploring the app without the hardware.
 - Added the ability to skip the tutorial after initial installation.
 - Fixed a bug where the firmware upgrading doesn’t work.
 - Fixed a bug where the app doesn’t recover from certain crashes.
 - Added a ripple animation to the shields list.
 - Fixed push notifications icon to the correct one.
 - Added the ability to open the context menu by clicking on 1Sheeld logo.
 - Added more validations to GPS and NFC shields.
 - Enhanced the camera implementation and made sure that the photo has correct focus and exposure.
 - Fixed a bug where data logger notifications don’t open on some devices.
 - Fixed a bug in the email shield where sending emails wasn't working in the background.
 - Increased overall app stability and fixed other major bugs.

## v1.5.0(151123) (November 23, 2015)
 - Make sure frames sent and received are correct.
 - Added copy selection feature to terminal shield.
 - Remove select all button in shields list.
 - Added uploading images feature to internet shield.
 - Added a progress bar while scanning for 1Sheelds.
 - Fixed a bug where the wrong image is uploaded in email and twitter shields.
 - Added a button in the camera shield to show the latest captured image.
 - Made the data logger shields output the logged file in a notification.
 - Revamped the login process in the email shield.
 - Added the ability to record sound from mic shield.
 - Fixed an error that prevented firmware upgrade on Android M devices.

## v1.4.1(151028) (October 28, 2015)
 - Fixed the connection bug on Android 6 devices.

## v1.4.0(150818) (August 18, 2015)
 - Added the GLCD shield.
 - Fixed an issue with the NFC shield.
 - Removed the Facebook shield temporarily due to a change in their permissions policy.
 - Increased overall app stability and fixed other major bugs.
 
## v1.3.1(150519) (May 19, 2015)
 - Fixed a bug where the app rating dialog appears each time a connection is made.
 - Fixed some bugs in the NFC shield.

## v1.3.0(150505) (May 5, 2015)
 - Added two new shields, NFC and color detector.
 - Enhanced the camera shield implementation.
 - Added a preview for the camera shield.
 - Removed RemoteOneSheeld shield.
 - Fixed selection callback for some shields.
 - Made the data logger shield save logged files on connection loss.
 - Increased overall app stability and fixed other major bugs.

## v1.2.0(150308) (March 29, 2015)
 - Added our most advanced - and most requested - shield ever, the Internet Shield.
 - Added an Android pattern shield.
 - Fixed a bug in Bluetooth connection status reporting.
 - Fixed major bugs related to sending frames to Arduino.
 - Revamped the camera shield implementation. (Now, it works in its own process.)
 - Queued frames if Arduino is in one of our callbacks.
 - Added the name of the connected 1Sheeld in the notifications.
 - Fixed bugs with the voice recognizer and Facebook shields.
 - Added a button to remember the list of selected shields.
 - Increased overall app stability.

## v1.1.3(141225) (December 25, 2014)
 - Fixed front camera orientation problem.
 - Added a feature to control 1Sheelds remotely through the internet (Remote 1Sheeld).
 - Report input pins on connection.
 - Notify the board with Bluetooth status.
 - Added a method to attach pictures to email shield.
 - Added methods to track keywords to Twitter shield.
 - Increased overall app stability and fixed other major bugs.
 - Prepared app for firmware upgrade.

## v1.1.2(140911) (September 11, 2014)
 - Fixed a major bug in Twitter shield.

## v1.1.1(140910) (September 10, 2014)
 - Fixed problems in data logger shield.
 - Added a automatic time stamp column in logged files.

## v1.1.0(140909) (September 9, 2014)
 - Added four new shields. (Data logger, voice recognition, text-to-speech and terminal)
 - Added 2 plugins for Tasker. (Condition and Action plugins)
 - Added the ability to post photos from Facebook and Twitter shields.
 - Added a method to send direct messages from Twitter shield.
 - Detect older versions of Arduino library.
 - Fixed the orientation of captured photos in camera shield.
 - Increased overall app stability.

## v1.0.5(140804) (August 4, 2014)
 - Fixed bugs in the connection dialog.
 - Fixed bugs in the music player shield.

## v1.0.4(140703) (July 3, 2014)
 - Made the camera shield use the highest resolution available.
 - Added the LCD shield.
 - Prepared the app for the next firmware upgrade.
 - Fixed other bugs and enhanced overall app stability.

## v1.0.3(140626) (June 26, 2014)
 - Made the camera shield use the highest resolution available.

OneSheeld/src/com/integreight/onesheeld/UIShield.java
package com.integreight.onesheeld;



public enum UIShield {
	
	LED_SHIELD(1, "LED",R.drawable.shields_activity_strip_13,R.drawable.shields_activity_strip_13_bw,R.drawable.shields_activity_small_strip_13,R.drawable.shields_activity_led_symbol, false),
	NOTIFICATION_SHIELD(2, "Notification",R.drawable.shields_activity_strip_1,R.drawable.shields_activity_strip_1_bw,R.drawable.shields_activity_small_strip_1,R.drawable.shields_activity_vibration_symbol, false),
	SEVENSEGMENT_SHIELD(3, "Seven Segment",R.drawable.shields_activity_strip_19,R.drawable.shields_activity_strip_19_bw,R.drawable.shields_activity_small_strip_19,R.drawable.shields_activity_seven_segment_symbol, false),
	BUZZER_SHIELD(4, "Buzzer",R.drawable.shields_activity_strip_9,R.drawable.shields_activity_strip_9_bw,R.drawable.shields_activity_small_strip_9,R.drawable.shields_activity_buzzer_symbol, false),
	MIC_SHIELD(5, "Mic",R.drawable.shields_activity_strip_15,R.drawable.shields_activity_strip_15_bw,R.drawable.shields_activity_small_strip_15,R.drawable.shields_activity_mic_symbol ,false),
	KEYPAD_SHIELD(6, "Keypad",R.drawable.shields_activity_strip_2,R.drawable.shields_activity_strip_2_bw,R.drawable.shields_activity_small_strip_2,R.drawable.shields_activity_keypad_symbol ,false),
	SLIDER_SHIELD(7, "Sliders",R.drawable.shields_activity_strip_3,R.drawable.shields_activity_strip_3_bw,R.drawable.shields_activity_small_strip_3,R.drawable.shields_activity_sliders_symbol ,false),
	LCD_SHIELD(8, "LCD",R.drawable.shields_activity_strip_4,R.drawable.shields_activity_strip_4_bw,R.drawable.shields_activity_small_strip_4,R.drawable.shields_activity_lcd_symbol ,false),
	MAGNETOMETER_SHIELD(9, "Magnetometer",R.drawable.shields_activity_strip_16,R.drawable.shields_activity_strip_16_bw,R.drawable.shields_activity_small_strip_16,R.drawable.shields_activity_magnetometer_symbol ,false),
	PUSHBUTTON_SHIELD(10, "Push Button",R.drawable.shields_activity_strip_12,R.drawable.shields_activity_strip_12_bw,R.drawable.shields_activity_small_strip_12,R.drawable.shields_activity_push_button_symbol ,false),
	TOGGLEBUTTON_SHIELD(11, "On/Off Button",R.drawable.shields_activity_strip_6,R.drawable.shields_activity_strip_6_bw,R.drawable.shields_activity_small_strip_6,R.drawable.shields_activity_push_button_symbol ,false),
	ACCELEROMETER_SHIELD(12, "Accelerometer",R.drawable.shields_activity_strip_21,R.drawable.shields_activity_strip_21_bw,R.drawable.shields_activity_small_strip_21,R.drawable.shields_activity_accelerometer_symbol ,false),
	FACEBOOK_SHIELD(13, "Facebook",R.drawable.shields_activity_strip_7,R.drawable.shields_activity_strip_7_bw,R.drawable.shields_activity_small_strip_7,R.drawable.shields_activity_facebook_symbol ,false),
	TWITTER_SHIELD(14, "Twitter",R.drawable.shields_activity_strip_17,R.drawable.shields_activity_strip_17_bw,R.drawable.shields_activity_small_strip_17,R.drawable.shields_activity_twitter_symbol ,false),
	GAMEDPAD_SHIELD(15, "Game Pad",R.drawable.shields_activity_strip_10,R.drawable.shields_activity_strip_10_bw,R.drawable.shields_activity_small_strip_10,R.drawable.shields_activity_gamepad_symbol ,false),
	INTERNET_SHIELD(16, "Internet",R.drawable.shields_activity_strip_5,R.drawable.shields_activity_strip_5_bw,R.drawable.shields_activity_small_strip_5,R.drawable.shields_activity_internet_symbol ,false),
	FOURSQUARE_SHIELD(17, "Foursquare",R.drawable.shields_activity_strip_8,R.drawable.shields_activity_strip_8_bw,R.drawable.shields_activity_small_strip_8,R.drawable.shields_activity_foursquare_symbol ,false),
	GPS_SHIELD(18, "GPS",R.drawable.shields_activity_strip_18,R.drawable.shields_activity_strip_18_bw,R.drawable.shields_activity_small_strip_18,R.drawable.shields_activity_gps_symbol ,false),
	SMS_SHIELD(19, "SMS",R.drawable.shields_activity_strip_20,R.drawable.shields_activity_strip_20_bw,R.drawable.shields_activity_small_strip_20,R.drawable.shields_activity_sms_symbol ,false),
	MUSICPLAYER_SHIELD(20, "Music Player",R.drawable.shields_activity_strip_11,R.drawable.shields_activity_strip_11_bw,R.drawable.shields_activity_small_strip_11,R.drawable.shields_activity_musicplayer_symbol ,false),
	GYROSCOPE_SHIELD(21, "Gyroscope",R.drawable.shields_activity_strip_14,R.drawable.shields_activity_strip_14_bw,R.drawable.shields_activity_small_strip_14,R.drawable.shields_activity_gyroscope_symbol ,false),
	FLASHLIGHT_SHIELD(22, "Flashlight",R.drawable.shields_activity_strip_22,R.drawable.shields_activity_strip_22_bw,R.drawable.shields_activity_small_strip_22,R.drawable.shields_activity_flashlight_symbol ,false);
	
	private int id;
	private String name;
	private int mainImageStripId;
	private int mainBWImageStripId;
	private int smallImageStripId;
	private int symbolId;
	private boolean mainActivitySelection;
	private static UIShield shieldsActivitySelection;
	private static boolean isConnected=false;
	
	public int getSymbolId() {
		return symbolId;
	}
	
//	public int getMainBWImageStripId() {
//		return mainBWImageStripId;
//	}
	public Boolean isMainActivitySelection() {
		return mainActivitySelection;
	}
	public static UIShield getShieldsActivitySelection() {
		return shieldsActivitySelection;
	}
	public static void setShieldsActivitySelection(UIShield selection) {
		shieldsActivitySelection = selection;
	}
	public void setMainActivitySelection(boolean toggleStatus) {
		this.mainActivitySelection = toggleStatus;
	}
	public int getMainImageStripId() {
		if(isConnected)return mainImageStripId;
		else return mainBWImageStripId;
	}
	
	public int getSmallImageStripId() {
		return smallImageStripId;
	}
	public String getName() {
		return name;
	}
	private UIShield(int id, String name ,int mainImageStripId,int mainBWImageStripId,int smallImageStripId ,int symbolId ,boolean mainActivitySelection) {
		this.id = id;
		this.name = name;
		this.mainImageStripId = mainImageStripId;
		this.mainBWImageStripId=mainBWImageStripId;
		this.mainActivitySelection = mainActivitySelection;
		this.symbolId = symbolId;
		this.smallImageStripId=smallImageStripId;
	
	
	}
	public int getId() {
		return id;
	}
	
	public static UIShield getItem(int id){
		switch (id) {
		case 1:    return LED_SHIELD;
		case 2:    return NOTIFICATION_SHIELD;
		case 3:    return SEVENSEGMENT_SHIELD;
		case 4:    return BUZZER_SHIELD;
		case 5:    return MIC_SHIELD;
		case 6:    return KEYPAD_SHIELD;
		case 7:    return SLIDER_SHIELD;
		case 8:    return LCD_SHIELD;
		case 9:    return MAGNETOMETER_SHIELD;
		case 10:    return PUSHBUTTON_SHIELD;
		case 11:    return TOGGLEBUTTON_SHIELD;
		case 12:    return ACCELEROMETER_SHIELD;  
		case 13:    return FACEBOOK_SHIELD;
		case 14:    return TWITTER_SHIELD;  
		case 15:    return GAMEDPAD_SHIELD;  
		case 16:    return INTERNET_SHIELD;  
		case 17:    return FOURSQUARE_SHIELD;  
		case 18:    return GPS_SHIELD;  
		case 19:    return SMS_SHIELD;  
		case 20:    return MUSICPLAYER_SHIELD;  
		case 21:    return GYROSCOPE_SHIELD;  
		case 22:    return FLASHLIGHT_SHIELD;  
		}
		return null;
	}

	public static boolean isConnected() {
		return isConnected;
	}

	public static void setConnected(boolean isConnected) {
		UIShield.isConnected = isConnected;
	}

	
	

	
}
